pkg/quota/evaluator/core/pods.go

```diff
@@ -254,6 +255,13 @@ func PodUsageFunc(obj runtime.Object) (api.ResourceList, error) {
 	if err != nil {
 		return api.ResourceList{}, err
 	}
+	// Only charge pod count for uninitialized pod.
+	if !initialization.IsInitialized(pod.Initializers) {
+		result := api.ResourceList{}
+		result[api.ResourcePods] = resource.MustParse("1")
```

Comment:
this changes behavior by moving object count above the `if !QuotaPod` check

Reply:
I'd expect this done unconditionally after the `!QuotaPod` check, then an early return `if !initialization.IsInitialized`, just like PV

Reply:
Good catch. Though probably `QuotaPod` is never effective. This evaluator [used](https://github.com/kubernetes/kubernetes/blob/v1.7.3/pkg/quota/evaluator/core/pods.go#L136) to only handle `Create`, and no one can create `Pod` with Phase=Succeeded || Phase == Failed

Reply:
> this changes behavior by moving object count above the if !QuotaPod check

Done. Moved to after `!QuotaPod`.

> I'd expect this done unconditionally after the !QuotaPod check, then an early return if !initialization.IsInitialized, just like PV

The pod evaluator summarizes all containers's request/limits and then invokes a [helper function](https://github.com/kubernetes/kubernetes/blob/master/pkg/quota/evaluator/core/pods.go#L277) to calculate the quota usage. Following the pvc pattern will require more change, so I keep it as is.